generator/src/lib.rs

```diff
@@ -1578,31 +1581,26 @@ pub fn derive_setters(_struct: &vkxml::Struct) -> Option<Tokens> {
         })
     });
 
-    let mut nexts = Vec::new();
     let extends_name = name_to_tokens(&format!("Extends{}", name));
-    if let Some(extends) = &_struct.extends {
-        for target in extends.split(',') {
-            let target = match target {
-                // https://github.com/KhronosGroup/Vulkan-Docs/pull/870
-                "VkPhysicalDeviceProperties" => "VkPhysicalDeviceProperties2",
-                x => x,
-            };
-            let target_ident = name_to_tokens(&format!("Extends{}", name_to_tokens(target)));
-            nexts.push(quote! {
-                unsafe impl #target_ident for #name {}
-            });
-        }
-    }
 
-    let next_function = if has_next {
+    let next_extends = _struct.extends.as_ref().map(|extends| {
+        let extends = extends
+            .split(',')
+            .find(|extend| root_create_info_names.contains(&extend.to_string()))
```

Comment:
When is this check necessary? Do there exist structs that extend non-root structs?

Reply:
> Do there exist structs that extend non-root structs?

Yes not all extend structs are root. I'll add a comment, and I to slightly rewrite this anyway.